### ISSUE TITLE
Docs: globalThis without ignores of no-var and no-namespace

### ISSUE THREAD

**Issue by karlhorky** [typescript-eslint/typescript-eslint#9582]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

A `typescript-eslint`-blessed solution for the problem of declaring `globalThis` properties without ignores:

```ts
declare global {
  var postgresSqlClient: Sql; // 💥 Unexpected var, use let or const instead. eslint (no-var)
}
```

`declare module globalThis` option:

```ts
declare module globalThis {
  let postgresSqlClient: Sql;
} // 💥 ES2015 module syntax is preferred over namespaces. eslint (@typescript-eslint/no-namespace)
```

### Affected URL(s)

- https://v8--typescript-eslint.netlify.app/rules/no-namespace/

**Comment by karlhorky** [comment]
Hm, maybe this would actually be better as a change to the `no-var` behavior, as suggested by a few users in the wild 🤔

- https://github.com/typescript-eslint/typescript-eslint/issues/3617
- https://github.com/typescript-eslint/typescript-eslint/issues/7941

**Comment by kirkwaiblinger** [comment]
I think it would be great for this to be documented, regardless of whether #7941 moves forward... 

The trouble with the `no-var` case is that there is presently no `@typescript-eslint/no-var` rule at all. Therefore, we don't have a https://typescript-eslint.io/rules/no-var page where we could document this behavior 🤔. Maybe it could go under the eslint FAQs, kind of like this blurb on `no-undef`? https://typescript-eslint.io/troubleshooting/faqs/eslint/#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors

**Comment by DanKaplanSES** [comment]
> Maybe it could go under the eslint FAQs, kind of like this blurb on no-undef?

**TL;DR:** I think adding this to the FAQ would help those that need it the least. 

I only have anecdotal evidence to back this up, but I believe most people experiencing this error will trust the linter and move on with their lives—or at least try to.

If there's lag between following the linter's advice and spotting the issue it created, I imagine "the linter told me to do the wrong thing" will be far from their mind, and you have to suspect the linter before you'll check its FAQ. 

I think most will research these first:

1. How to add a typed property to `window`, `global`, and/or `globalThis`
2. `window`, `global`, and/or `globalThis` in general
3. Declaration files
4. The `declare` keyword
5. The typescript server
6. The vscode issue tracker
7. etc.

At least, this is the path I went down until I circled back and dug my heels into `1.` Only then did I realize the linter took me down the wrong path. It was quite a journey, which is why I think modifying the rule would be preferable. 

That said, I do think putting this in the FAQ is an objective improvement over the current situation.

Just my two cents.

**Comment by kirkwaiblinger** [comment]
> > I think it would be great for this to be documented, regardless of whether #7941 moves forward...
> 
> Agreed!
> 
> > Maybe it could go under the eslint FAQs, kind of like this blurb on no-undef?
> 
> **TL;DR:** I'm concerned the FAQ would help those that need it the least.
> 
> I only have anecdotal evidence to back this up, but I believe most people experiencing this error will trust the linter and move on with their lives—or at least try to.
> 
> If there's lag between following the linter's advice and spotting the issue it created, I imagine "the linter told me to do the wrong thing" will be far from their mind, and you have to suspect the linter before you'll check its FAQ.
> 
> I think most will research these first:
> 
> 1. How to add a typed property to `window`, `global`, and/or `globalThis`
> 2. `window`, `global`, and/or `globalThis` in general
> 3. Declaration files
> 4. The `declare` keyword
> 5. The typescript server
> 6. The vscode issue tracker
> 7. etc.
> 
> At least, this is the path I went down until I circled back and dug my heels into `1.` Only then did I realize the linter took me down the wrong path. It was quite a journey, which is why I think ~modifying~ creating the rule would be preferable.
> 
> That said, I do think putting this in the FAQ is an objective improvement over the current situation.
> 
> Just my two cents.

***

I agree with essentially every word here. Opening this to PRs for the FAQs. As for changing behavior, the best way for anyone to advocate that change for now is to engage in https://github.com/typescript-eslint/typescript-eslint/issues/7941, by adding a 👍 or ❤️, or commenting with additional data.

**Comment by karlhorky** [comment]
@JoshuaKGoldberg Thanks for the documentation PR at #9865, looks great 🙌
